src/baldr/pathlocation.cc

```diff
@@ -61,5 +63,172 @@ bool PathLocation::shares_edges(const PathLocation& other) const {
   return true;
 }
 
+void PathLocation::toPBF(const PathLocation& pl, valhalla::Location* l, baldr::GraphReader& reader) {
```

Comment:
Is it possible to separate this PR into multiple? For example
- changes like this one
- forward declarations
- includes reordering

Reply:
I know it's big and again, I'm not very eager to have this merged. I shouldn't have started it, but once I did, there was no stopping :D 

Given the amount of changes, I wouldn't go through that again tbh.